src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java

```diff
@@ -127,17 +133,16 @@ public CqlSession getSession() {
         && !stargateRequestInfo.getCassandraToken().orElseThrow().equals(fixedToken)) {
       throw new UnauthorizedException("Unauthorized");
     }
-    return sessionCache.get(getSessionCacheKey(), this::getNewSession);
+    return sessionCache.get(getSessionCacheKey());
   }
 
   /**
    * Default token which will be used by the integration tests. If this property is set, then the
    * token from the request will be compared with this to perform authentication.
    */
   private String getFixedToken() {
-    return operationsConfig.databaseConfig().fixedToken().isPresent()
-        ? operationsConfig.databaseConfig().fixedToken().get()
-        : null;
+    Optional<String> fixedToken = operationsConfig.databaseConfig().fixedToken();
+    return fixedToken != null && fixedToken.isPresent() ? fixedToken.get() : null;
```

Comment:
Should we ever get `null` for something that should be `Optional`? (usually not)